tests/AbstractTestCase.php

```diff
@@ -41,4 +52,64 @@ public function createApplication()
 
         return $app;
     }
+
+    /**
+     * Sign in an user if it's the case.
+     *
+     * @param User|null $user
+     *
+     * @return AbstractTestCase
+     */
+    protected function signIn(User $user = null)
+    {
+        if ($user) {
```

Comment:
Should this not just be something like `$this->user = $user ?: $this->createUser();`?

Seems a bit... Messy. Though I'm unsure on the style guide for this. @GrahamCampbell?